src/datatypes/src/data_type.rs

```diff
@@ -68,7 +68,6 @@ pub enum ConcreteDataType {
 
     // Date and time types:
     Date(DateType),
-    DateTime(DateTimeType),
```

Comment:
This will be a breaking change.

@v0y4g3r @fengjiachun Do we need to consider existing tables with DateTime type? People may not be able to open these tables anymore.

Reply:
Yes, this is a breaking change, but I still believe we can remove the `DateTime` type because even if users are using this type, the stored data is incorrect. We can mark this PR as `breaking`.